packages/blocks/github-pr-overview/src/entity-aggregations.ts

```diff
@@ -0,0 +1,257 @@
+import {
+  BlockProtocolAggregateEntitiesFunction,
+  BlockProtocolAggregateEntitiesResult,
+} from "blockprotocol";
+import { uniqBy } from "lodash";
+
+import {
+  GithubIssueEvent,
+  GithubPullRequest,
+  PullRequestIdentifier,
+  GithubReview,
+  isDefined,
+} from "./types";
+
+const ITEMS_PER_PAGE = 100;
+
+export const getPrs = (
+  githubPullRequestTypeId: string,
+  aggregateEntities?: BlockProtocolAggregateEntitiesFunction,
+  accountId?: string | null,
+  pageNumber?: number,
+  selectedPullRequest?: PullRequestIdentifier,
+): Promise<BlockProtocolAggregateEntitiesResult<GithubPullRequest> | void> => {
+  if (!aggregateEntities) {
+    return new Promise<void>(() => {});
+  }
+
+  const res = aggregateEntities({
+    accountId,
+    operation: {
+      entityTypeId: githubPullRequestTypeId,
+      pageNumber,
+      itemsPerPage: ITEMS_PER_PAGE,
+      multiFilter: {
+        operator: "AND",
+        filters:
+          selectedPullRequest !== undefined
+            ? [
+                {
+                  field: "html_url",
+                  operator: "CONTAINS",
+                  value: `${selectedPullRequest.repository}/pull/${selectedPullRequest.number}`,
+                },
+              ]
+            : [],
+      },
+      multiSort: [
+        {
+          field: "url",
+          desc: true,
+        },
+      ],
+    },
+  });
+
+  return res;
+};
+
+export const collectPrsAndSetState = (
+  githubPullRequestTypeId: string,
+  aggregateEntities: BlockProtocolAggregateEntitiesFunction | undefined,
+  accountId: string | null | undefined,
+  numPages: number,
+  setState: (x: any) => void,
+) => {
+  const results = Array(numPages)
+    .fill(undefined)
+    .map((_, pageNumber) =>
+      getPrs(githubPullRequestTypeId, aggregateEntities, accountId, pageNumber),
+    );
+
+  Promise.all(results)
```

Comment:
Tweak for clarity, not important. Applies elsewhere.

```suggestion
  const promises = Array(numPages)
    .fill(undefined)
    .map((_, pageNumber) =>
      getPrs(githubPullRequestTypeId, aggregateEntities, accountId, pageNumber),
    );

  Promise.all(promises)
```

Reply:
Applied